From d1dea3c2bcf1e22baab39e2c0b8ca59db8a5bc37 Mon Sep 17 00:00:00 2001
From: Randy MacLeod <Randy.MacLeod@windriver.com>
Date: Mon, 13 May 2019 17:12:21 -0400
Subject: [PATCH 1/2] valgrind: filter_xml_frames: do not filter /usr

filter_xml_frames is intended to filter system paths under
'/usr' that vary from platform to platform. In the ptest case
for Yocto's valgrind, the ptest executables are placed under:
   /usr/lib/valgrind/ptest
and if these frames are filtered out, then the 'drd' tests fail
the comparision between exepected and actual output.

Changing this filter allows the comparison to succeed without
any negative impact.

Upstream-Status: Inappropriate [Yocto specific]

Signed-off-by: Randy MacLeod <Randy.MacLeod@windriver.com>
---
 tests/filter_xml_frames | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/filter_xml_frames b/tests/filter_xml_frames
index f1fa3f070..04fbc82de 100755
--- a/tests/filter_xml_frames
+++ b/tests/filter_xml_frames
@@ -47,7 +47,8 @@ while (my $line = <>)
         $has_function_name = 1 if ($line =~ /<fn>/);
 # This may require tweaking; currently  /lib and /usr/lib are matched
         $has_system_obj = 1 if ($line =~ /<obj>\/lib/);
-        $has_system_obj = 1 if ($line =~ /<obj>\/usr\/lib/);
+# for Yocto, skip /usr since tests are under /usr/lib/...
+#        $has_system_obj = 1 if ($line =~ /<obj>\/usr\/lib/);
     }
 }
 
-- 
2.17.0

